pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go

```diff
@@ -182,76 +183,144 @@ func TestNewNSTemplateTier(t *testing.T) {
+				// error occurs when fetching the content of the 'advanced-code.yaml' template
+				return []byte("invalid"), nil // return an invalid YAML represention of a Template
+			})
+			namespace := "host-operator-" + uuid.NewV4().String()[:7]
+			templatesByTier, err := loadTemplatesByTiers(fakeAssets)
+			require.NoError(t, err)
+			advancedTemplates := templatesByTier["advanced"]
+			// when
+			_, err = newNSTemplateTier(decoder, namespace, "advanced", *advancedTemplates)
+			// then
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "unable to generate 'advanced' NSTemplateTier manifest: couldn't get version/kind; json parse error")
+		})
+	})
+}
+
+func TestNewTierTemplate(t *testing.T) {
+
+	s := scheme.Scheme
+	decoder := serializer.NewCodecFactory(s).UniversalDeserializer()
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+
+	t.Run("ok", func(t *testing.T) {
+
+		t.Run("with prod assets", func(t *testing.T) {
+			// given
+			assets := NewAssets(AssetNames, Asset)
+			// uses the `Asset` funcs generated in the `pkg/templates/nstemplatetiers/` subpackages
+			templatesByTier, err := loadTemplatesByTiers(assets)
+			require.NoError(t, err)
+			namespace := "host-operator-" + uuid.NewV4().String()[:7]
+			// when
+			tierTmpls, err := newTierTemplates(decoder, namespace, templatesByTier)
+			// then
+			require.NoError(t, err)
+			require.Len(t, tierTmpls, 11) // current count of template files
+			decoder := serializer.NewCodecFactory(s).UniversalDeserializer()
+
+			resourceNameRE, err := regexp.Compile(`[a-z0-9\.-]+`)
+			require.NoError(t, err)
+			for _, actual := range tierTmpls {
+				t.Run(actual.Name, func(t *testing.T) {
+					assert.Equal(t, namespace, actual.Namespace)
+					assert.True(t, resourceNameRE.MatchString(actual.Name)) // verifies that the TierTemplate name complies with the DNS-1123 spec
+					assert.NotEmpty(t, actual.Spec.Revision)
+					assert.NotEmpty(t, actual.Spec.TierName)
+					assert.NotEmpty(t, actual.Spec.Type)
+					assert.NotEmpty(t, actual.Spec.Template)
+					switch actual.Spec.Type {
+					case "dev", "code", "stage":
+						assertNamespaceTemplate(t, decoder, actual.Spec.Template, actual.Spec.TierName, actual.Spec.Type)
+					case "clusterResources":
+						assertClusterResourcesTemplate(t, decoder, actual.Spec.Template, actual.Spec.TierName)
+					default:
+						t.Errorf("unexpected kind of template: '%s'", actual.Spec.Type)
+					}
```

Comment:
this could be a bit shorter:
```suggestion
					if actual.Spec.Type == "clusterResources" {
						assertClusterResourcesTemplate(t, decoder, actual.Spec.Template, actual.Spec.TierName)
					} else {
						assertNamespaceTemplate(t, decoder, actual.Spec.Template, actual.Spec.TierName, actual.Spec.Type)
					}
```
but your version is fine too ;-)

Reply:
I'll keep it as-is. It supports the case where the type is unexpected and thus is shows that we handle expect `dev`, "code` and `stage` and nothing else for the namespaces.

Reply:
fair enough :wink: :+1: